core/sr-arithmetic/benches/bench.rs

```diff
@@ -0,0 +1,76 @@
+use criterion::{Criterion, criterion_group, criterion_main};
+use sr_arithmetic::biguint::{BigUint, Single};
+
+fn random_big_uint(size: usize) -> BigUint {
+	use rand::Rng;
+	let mut rng = rand::thread_rng();
+	let digits: Vec<_> = (0..size).map(|_| rng.gen_range(0, Single::max_value())).collect();
+	BigUint::from_limbs(&digits)
+}
+
+fn bench_addition_2_digit(c: &mut Criterion) {
```

Comment:
I think now with criterion you can create a vector of limb sizes that you want `vec![2, 4, 6, 8, 10, ...]` and pass that and then configure the benchmark to generate numbers based on that... I have to look it up my memory of criterion is a bit foggy now but I am pretty sure criterion is much much more flexible and can probably circumvent this ugly pattern of `bench_size_x()`, `bench_size_y()` that I added initially.

Reply:
Done :^)

Reply:
ahh nice I didn't even have to look it up :D thanks.